spec/arrays.dd

```diff
@@ -456,18 +456,30 @@ double[][] matrix;
 
 $(SPEC_RUNNABLE_EXAMPLE_COMPILE
 ---------
-double[3][3] matrix;
+double[6][3] matrix;
 ---------
 )
 
-        $(P declares a rectangular matrix with 3 rows and 3 columns, all contiguously in memory. In
+        $(P declares a rectangular matrix with 3 rows and 6 columns, all contiguously in memory. In
         other languages, this would be called a multidimensional array and be declared as:
         )
 
 ---------
-double matrix[3,3];
+double matrix[3,6];
 ---------
 
+        $(P Note that dimensions and indices appear in opposite orders. Dimensions in the
+        $(RELATIVE_LINK2 declarations, declaration) are read right to left whereas indices are read
+        left to right:
+        )
+
+$(SPEC_RUNNABLE_EXAMPLE_COMPILE
+---------
+double[6][3] matrix;
+matrix[2][5] = 3.14; // Assignment to bottom right element.
```

Comment:
I have deliberately avoided use of the words "column" and "row" because it differs whether your mind is set to column-major or row-major mode... In both modes it is the bottom right.

How about:
```d
int[6][3] matrix;
matrix[2][5] = 42; // Assignment to bottom right element.
writeln(matrix); // [[0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 0], [0, 0, 0, 0, 0, 42]]

static assert(!__traits(compiles, matrix[5][2])); // Array index out of bounds.
```

I know those words do appear earlier, I'm hesitant to remove them there too. We could print the matrix there as well, but without initialization you'd get all `nan`s. Initialization is not as clean as ordinary array init, the best I can come up with is:
```d
matrix[0][] = 0.0;
matrtix[1..$][] = matrix[0];
```
We could switch from `double` to `int` like above, but floating point matrices are more common.

Reply:
> Initialization is not as clean as ordinary array init, the best I can come up with is:
> 
> ```d
> matrix[0][] = 0.0;
> matrtix[1..$][] = matrix[0];
> ```

Try this on for size:

```d
double[6][3] matrix = 0; /* sets all elements to 0 */
```

I was pretty sure that that's a documented feature, but I can't find it in spec now, so I might be imagining things. It works, though.

Reply:
> ```d
> double[6][3] matrix = 0; /* sets all elements to 0 */
> ```
> 
> I was pretty sure that that's a documented feature, but I can't find it in spec now, so I might be imagining things. It works, though.

Thanks for pointing that out. All the more reason to demonstrate that here then. I'll make another pass.